lms/djangoapps/philu_api/views.py

```diff
@@ -190,10 +190,8 @@ def assign_user_badge(request):
                             status=status.HTTP_403_FORBIDDEN)
 
     try:
-        UserBadge.assign_badge(user_id=user_id,
-                               badge_id=badge_id,
-                               community_id=community_id)
-        return JsonResponse({'success': True})
+        assigned = UserBadge.assign_badge(user_id=user_id, badge_id=badge_id, community_id=community_id)
+        return JsonResponse({'success': True}, status=status.HTTP_201_CREATED if assigned else status.HTTP_409_CONFLICT)
```

Comment:
In case of team badges, this code is inconsistent, and should be updated.

`HTTP_409_CONFLICT` is not appropriate here. 409 is for conflicts distributed in nature